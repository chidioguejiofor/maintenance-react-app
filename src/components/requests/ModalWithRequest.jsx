
import React from 'react';
import { Modal, Button, Container, Header, Label,
  Image, Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { getColor, getIconType,
  capitalizeWord } from '../../helpers/requestHelper';

const ModalWithRequest = ({
  title, description, location, status, date, image
}) => (
  <span id="modal-with-request">
    <Modal

      size="small"
      trigger={(
        <Button
          size="small"
          content="Info"
          icon="info"
        />
      )}
    >
      <Modal.Header>{title.toUpperCase()}</Modal.Header>
      <Modal.Content>
        <Container>

          <Image centered src={image} size="big" />
          <div className="request">
            <Header size="large">Created On </Header>
            <p>{`${new Date(date).toDateString()}`}</p>
            <Header size="large">Description</Header>
            <p>{description}</p>
            <Header size="large">Location </Header>
            <p>{location}</p>
            <Label size="large">
              <Icon
                name={getIconType(status)}
                color={getColor(status)}
              />
              {capitalizeWord(status)}
            </Label>
          </div>

        </Container>
      </Modal.Content>
    </Modal>

  </span>
);


ModalWithRequest.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,

};


export default ModalWithRequest;
