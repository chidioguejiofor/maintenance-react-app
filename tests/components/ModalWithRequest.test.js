import React from 'react';
import { shallow } from 'enzyme';
import { Modal } from 'semantic-ui-react';

import chai from 'chai';
import ModalWithRequest from
  '../../src/components/requests/ModalWithRequest';


const sampleProps = {
  date: '2018-10-06T00:00:00.000Z',
  title: 'I am the one that just updated',
  description: 'descriptoin 2',
  location: 'locatrion',
  image: 'sampleImage.png',
  status: 'pending',
  message: ''

};
const getShallowObj = (props = sampleProps) => shallow(
  <ModalWithRequest {...props} />
);
describe('Testing the RequestModalWithForm Component', () => {
  describe('the content it renders', () => {
    it('should render a Modal', () => {
      const shallowObj = getShallowObj();
      chai.expect(shallowObj.find(Modal).length)
        .to.equal(1);
    });
    it('should match snapshot', () => {
      const shallowObj = getShallowObj();
      expect(shallowObj).toMatchSnapshot();
    });
  });
});
