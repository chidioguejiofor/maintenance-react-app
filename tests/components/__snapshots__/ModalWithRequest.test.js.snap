// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the RequestModalWithForm Component the content it renders should match snapshot 1`] = `
<span
  id="modal-with-request"
>
  <Modal
    centered={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    size="small"
    trigger={
      <Button
        as="button"
        content="Info"
        icon="info"
        role="button"
        size="small"
      />
    }
  >
    <ModalHeader>
      I AM THE ONE THAT JUST UPDATED
    </ModalHeader>
    <ModalContent>
      <Container>
        <Image
          as="img"
          centered={true}
          size="big"
          src="sampleImage.png"
          ui={true}
        />
        <div
          className="request"
        >
          <Header
            size="large"
          >
            Created On 
          </Header>
          <p>
            Sat Oct 06 2018
          </p>
          <Header
            size="large"
          >
            Description
          </Header>
          <p>
            descriptoin 2
          </p>
          <Header
            size="large"
          >
            Location 
          </Header>
          <p>
            locatrion
          </p>
          <Label
            size="large"
          >
            <Icon
              as="i"
              color="orange"
              name="hourglass half"
            />
            Pending
          </Label>
        </div>
      </Container>
    </ModalContent>
  </Modal>
</span>
`;
